gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/test/java/io/gravitee/am/gateway/handler/common/vertx/web/handler/AuthenticationFlowHandlerTest.java

```diff
@@ -721,65 +879,76 @@ public void shouldRedirectToMFAChallengePage_adaptiveMFA() throws Exception {
             // set client
             Client client = new Client();
-            client.setFactors(Collections.singleton("factor-1"));
+            setFactors(client);
             rc.put(ConstantKeys.CLIENT_CONTEXT_KEY, client);
-            MFASettings mfaSettings = new MFASettings();
             final RememberDeviceSettings rememberDevice = new RememberDeviceSettings();
             rememberDevice.setActive(true);
             mfaSettings.setRememberDevice(rememberDevice);
             rc.session().put(DEVICE_ALREADY_EXISTS_KEY, true);
-            mfaSettings.setStepUpAuthenticationRule("{#request.params['scope'][0].contains('write')}");
-            mfaSettings.setAdaptiveAuthenticationRule("{#context.attributes['geoip']['country_iso_code'] == 'FR'}");
+
+            var stepUpAuthenticationRule = "{#request.params['scope'][0].contains('write')}";
+            var stepUpAuthentication = new StepUpAuthenticationSettings();
+            stepUpAuthentication.setActive(true);
+            stepUpAuthentication.setStepUpAuthenticationRule(stepUpAuthenticationRule);
+            mfaSettings.setStepUpAuthentication(stepUpAuthentication);
+
             rc.put(ConstantKeys.GEOIP_KEY, new JsonObject().put("country_iso_code", "FR").getMap());
             client.setMfaSettings(mfaSettings);
             // set user
             EnrolledFactor enrolledFactor = new EnrolledFactor();
-            enrolledFactor.setFactorId("factor-1");
+            enrolledFactor.setFactorId(FACTOR_ID);
             enrolledFactor.setStatus(ACTIVATED);
             io.gravitee.am.model.User endUser = new io.gravitee.am.model.User();
             endUser.setFactors(Collections.singletonList(enrolledFactor));
             rc.getDelegate().setUser(new User(endUser));
-            rc.session().put(ConstantKeys.STRONG_AUTH_COMPLETED_KEY, true);
+            rc.session().put(STRONG_AUTH_COMPLETED_KEY, true);
             rc.next();
         });
 
         testRequest(
-                HttpMethod.GET, "/login?scope=write",
-                null,
-                resp -> {
-                    String location = resp.headers().get("location");
-                    assertNotNull(location);
-                    assertTrue(location.endsWith("/mfa/challenge?scope=write"));
-                },
-                HttpStatusCode.FOUND_302, "Found", null);
+                HttpMethod.GET, "/login?scope=read",
+                HttpStatusCode.OK_200, "OK");
     }
 
     @Test
     public void shouldContinue_adaptiveMFA_with_step_up_false_strong_auth_true_device_known() throws Exception {
         router.route().order(-1).handler(rc -> {
+            String rule = "{#context.attributes['geoip']['country_iso_code'] == 'FR'}";
+            EnrollSettings enrollSettings = createEnrollSettings(true, false, MfaEnrollType.REQUIRED, 0, "");
+            ChallengeSettings challengeSettings = createChallengeSettings(true, MfaChallengeType.RISK_BASED, rule);
```

Comment:
**todo** : based on the test case name, the test is related to adaptive MFA but the Challenge type is RISK_BASED, should be CONDITIONAL